include/pinmux.h

```diff
@@ -12,6 +12,12 @@
 #ifndef __INCLUDE_PINMUX_H
 #define __INCLUDE_PINMUX_H
 
+#if defined(CONFIG_PINCTRL)
+/* PINMUX is provided by PINCTRL */
+#include <pinctrl.h>
+#else
```

Comment:
pinmux.h will need to go away in the same PR as well. All drivers/boards will need to switched to pinctrl.

Reply:
Sorry it is not possible to do a all at once switch here as a proof of concept. You may keep the pull request in the DNM state as long as there are not all drivers switched. During the transition phase both worlds have to live in parallel. Before one can think of fully switching pinctrl drivers for all architectures have to be written and !!!TESTTED!!!. Even in Linux this is an ongoing process. At the end pinmux should go away and be replaced by pinctrl.